tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/GroupingTest.java

```diff
@@ -46,7 +45,7 @@ public void testSingleGroupExecution() throws BallerinaTestException {
         String output = balClient.runMainAndReadStdOut("test", args,
                 new HashMap<>(), projectPath, false);
         if (!output.contains(msg)) {
-            AssertionUtils.assertForTestFailures(output, "single group test execution failure");
+            throw new BallerinaTestException("Test failed due to single group test execution failure.");
```

Comment:
Isn't it better to have an `Assert.fail` instead of throwing a ballerina exception?

Reply:
+1 will check on this.

Reply:
Fixed with c90803f16da6c2e757a99f5ed397a8e422f040d8